homeassistant/components/media_player/clementine.py

```diff
@@ -0,0 +1,223 @@
+            else:
+                self._state = STATE_PAUSED
+
+            if client.last_update and (time.time() - client.last_update > 40):
+                self._state = STATE_OFF
+
+            self._volume = float(client.volume) if client.volume else 0.0
+
+            if client.current_track:
+                self._track_id = client.current_track['track_id']
+                self._track_name = client.current_track['title']
+                self._track_artist = client.current_track['track_artist']
+                self._track_album_name = client.current_track['track_album']
+
+        except:
+            self._state = STATE_OFF
+            raise
+
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        return self._state
+
+    @property
+    def volume_level(self):
+        """Volume level of the media player (0..1)."""
+        return self._volume / 100.0
+
+    @property
+    def source(self):
+        """Return  current source name."""
+        source_name = "Unknown"
+        client = self._client
+        if client.active_playlist_id in client.playlists:
+            source_name = client.playlists[client.active_playlist_id]['name']
+        return source_name
+
+    @property
+    def source_list(self):
+        """List of available input sources."""
+        source_names = [s["name"] for s in self._client.playlists.values()]
+        return source_names
+
+    def select_source(self, source):
+        """Select input source."""
+        client = self._client
+        sources = [s for s in client.playlists.values() if s['name'] == source]
+        if len(sources) == 1:
+            client.change_song(sources[0]['id'], 0)
+
+    @property
+    def media_content_type(self):
+        """Content type of current playing media."""
+        return MEDIA_TYPE_MUSIC
+
+    @property
+    def media_title(self):
+        """Title of current playing media."""
+        return self._track_name
+
+    @property
+    def media_artist(self):
+        """Artist of current playing media, music track only."""
+        return self._track_artist
+
+    @property
+    def media_album_name(self):
+        """Album name of current playing media, music track only."""
+        return self._track_album_name
+
+    @property
+    def supported_features(self):
+        """Flag media player features that are supported."""
+        return SUPPORT_CLEMENTINE
+
+    @property
+    def media_image_hash(self):
+        """Hash value for media image."""
+        if self._client.current_track:
+            return self._client.current_track['track_id']
+
+        return None
+
+    @asyncio.coroutine
+    def async_get_media_image(self):
+        """Fetch media image of current playing image."""
+        if self._client.current_track:
+            image = bytes(self._client.current_track['art'])
```

Comment:
Your stay inside Event Loop. Your Access only of cached value? You can not access to a file or network resource on this Point.